test/clubbing_top_level.txt

```diff
@@ -0,0 +1 @@
+<table border="1"><tr><th>aaa</th><th>ccc</th></tr><tr><td>1</td><td>2</td></tr><tr><td>bbb</td><td>2</td></tr></table>
```

Comment:
I agree on a clubbing switch. I'll add that later when I get time - or you can if you want, it's really simple I think. What behaviour do you expect on your input? To me it looks correct.